### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by AbdollahRaafat**
In this branch, we built a recipe app that keeps track of all your recipes, ingredients, and inventory. It allows you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe.
Also, since sharing recipes is an important part of cooking the app should allow you to make them public so anyone can access them.

- Use controllers to handle requests and render empty views.
-  Use params from browser requests in a safe way.
-  Use preprocessed HTML files with embedded Ruby code.
-  Use layouts and templates for shared content.
-  Use database migration files to maintain database schema.
-  Use validations for models.
-  Secure app from n+1 problems.
-  Write SQL queries with ActiveRecord.
-  Set up associations between models.
-  Build a web app that requires the user to log in.
-  Use Devise gem for authentication.
-  Limit access to web app resources based on authorization rules.
-  Resolve errors

Participants (2)
@AbdollahRaafat 
@AngomRanjan

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  def create
+    recipe = Recipe.new(recipe_params)
+    recipe.user = current_user
+
+    if recipe.save
+      redirect_to recipe, notice: 'Recipe was successfully created.'
+    else
+      redirect_to new_recipe_path, notice: 'There was an error creating the recipe.'
+    end
+  end
```
Comment (muneebulrehman) [inline app/controllers/recipes_controller.rb:26]: When trying to create recipes, I was unable to do so as it was throwing this error on the screen. ⬇️
image
You can also watch [this video](https://www.loom.com/share/09be395ccc8744d986d288554cebd7f5) where I demonstrated what exactly is happening.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+end
```
Comment (muneebulrehman) [inline app/controllers/users_controller.rb:5]: You have not implemented a logout functionality for a user to log out and log in as a different user. Kindly implement this functionality and submit it for a re-review. 
You can also refer to [this video](https://www.loom.com/share/09be395ccc8744d986d288554cebd7f5) for more information.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good documentation

## Required Changes ♻️
_Check the comments under the review._

- [ ] You also need to write the tests for views ( recipe page, the food page, shopping list, public recipes). 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @AbdollahRaafat, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Recipe features ✔
- Styling ✔
- Flash notification ✔
- No linter errors ✔
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**